packages/reactive_ble_platform_interface/lib/src/model/association_info.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:functional_data/functional_data.dart';
+import 'package:meta/meta.dart';
+
+part 'association_info.g.dart';
+
+//ignore_for_file: annotate_overrides
+
+@immutable
+@FunctionalData()
+class AssociationInfo extends $AssociationInfo {
```

Comment:
is DeviceAssociationInfo a better name? Then we can call the property `macAddress`. Also let's add docs to the class